exist-core/src/main/java/org/exist/source/ClassLoaderSource.java

```diff
@@ -65,7 +65,7 @@ private static URL sourceToUrl(String source) throws IOException {
     public String type() {
         final String protocol = url.getProtocol();
         final String host = url.getHost();
-        if (protocol.equals("file") && (host == null || host.length() == 0 || "localhost".equals(host) || "127.0.0.1".equals(host))) {
+        if (protocol.equals("file") && (host == null || host.isEmpty() || "localhost".equals(host) || "127.0.0.1".equals(host))) {
```

Comment:
Do we actually need the two null and empty checks of host?
```java
if (protocol.equals("file") && ("localhost".equals(host) || "127.0.0.1".equals(host))) {
```

Reply:
The host==null check protects the .isEmpty() in case of a null value. If no host is provided this means 'File'. The URL class does not mention possible values so we should handle this.

Reply:
think so yes